feddlib/core/FE/Domain_decl.hpp

```diff
@@ -558,6 +560,15 @@ class Domain {
    /// @param name export suffix to identify flags
    void exportElementOrientation(string name = "default");
 
+   //Add here an additional function which sets for each FiniteElement its normal to the computed value !CAUTION preProcessMesh has to be called first if not normal calculation does not always compute the outward normal
```

Comment:
Can we check for this? i.e. check if the mesh has been preprocessed and terminate with an error message if not?

Reply:
Okey I would  add in the domain class an boolean preProcessedMesh_ which is set to true when the function 
template <class SC, class LO, class GO, class NO>
void Domain<SC,LO,GO,NO>::preProcessMesh(bool correctSurfaceNormals, bool correctElementOrientation){ 

    if(correctSurfaceNormals)
        mesh_->correctNormalDirections();

    if(correctElementOrientation)
        mesh_->correctElementOrientation();


}
is called with two true values, or would you suggest?

Reply:
If that is what `setSurfaceNormalsForFE()` needs to work correctly, then that sounds good. 
Or do you think this check is not necessary?